bin/list_rc_log.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env python3
+
+import argparse
+import datetime
+import email
+import subprocess
+import sys
+
+import dateutil.parser
+import pytz
+
+
+def is_greg_request(m):
+    if 'X-KernelTest-Branch' in m and 'in-reply-to' not in m:
+        return True
+    return False
+
+
+def is_beyond_time_search(m, limit):
+    dt = msg_get_dt(m)
+    if dt < limit:
+        return True
+    return False
+
+
+def msg_get_dt(m):
+    if 'date' in m:
+        return dateutil.parser.parse(m['date'])
+    else:
+        return None
+
+
+def get_version(m):
+    sub = m['subject']
+    if not sub.endswith("-stable review"):
+        return False
+    if not sub.startswith("["):
+        return False
+    sub1 = sub.replace('-stable review', '')
+    return sub1[sub1.rfind(' ')+1:]
+
+
+def get_review_requests(dt_limit):
+    print("* Looking for review requests...")
+    fg = {}
+    x = 0
+    while True:
+        headn = "HEAD~%d:m" % x
+
+        # Get the email message proper
+        email_bytes = subprocess.check_output(["/usr/bin/git", "show", headn])
```

Comment:
I assume this works but have you checked gitpython or python-git? They allow to access git details without calling external process

Reply:
I had not tried, but I did now, with gitpython (looks super handy). As I was converting this bit by bit, I ran it and noticed no speed improvements. By the end of the conversion it was 15% slower (!), with the biggest impact reading the blob and converting that to an email message. That has the biggest impact as it needs to look for "In-Reply-To" headers for each and every message.

I'll keep gitpython installed as it looks like it will become useful next time.